src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleTransferServiceTest.java

```diff
@@ -134,4 +155,46 @@ public void populateDataIfComingFromCT() {
 
     }
 
+    @Test
+    public void validateCasesBeforeTransfer() {
+        List<String> errors = new ArrayList<>();
+        var caseData = new CaseData();
+        caseData.setEthosCaseReference("245004/2020");
+
+        // Un cleared BF Action
+        var bfActionType = new BFActionType();
+        bfActionType.setDateEntered("2020-11-11");
+        var bfActionTypeItem = new BFActionTypeItem();
+        bfActionTypeItem.setValue(bfActionType);
+        caseData.setBfActions(List.of(bfActionTypeItem));
+
+        // Listed hearing
+        var dateListedType = new DateListedType();
+        dateListedType.setHearingStatus(HEARING_STATUS_LISTED);
+        var dateListedTypeItem = new DateListedTypeItem();
+        dateListedTypeItem.setValue(dateListedType);
+        var hearingType = new HearingType();
+        hearingType.setHearingDateCollection(List.of(dateListedTypeItem));
+        var hearingTypeItem = new HearingTypeItem();
+        hearingTypeItem.setValue(hearingType);
+        caseData.setHearingCollection(List.of(hearingTypeItem));
+
+        var submitEvent = new SubmitEvent();
+        submitEvent.setCaseData(caseData);
+        submitEvent.setState(ACCEPTED_STATE);
+        submitEvent.setCaseId(1232121232);
+
+        when(excelReadingService.readExcel(anyString(), anyString(), anyList(), any(), any()))
+                .thenReturn(multipleObjects);
+        when(singleCasesReadingService.retrieveSingleCases(anyString(), anyString(), anyList(), anyString()))
+                .thenReturn(new ArrayList<>(Collections.singletonList(submitEvent)));
+        doCallRealMethod().when(caseTransferService).validateCase(isA(CaseData.class), anyList());
+
+        caseTransferService.validateCase(caseData, errors);
+        assertEquals(2, errors.size());
+        assertEquals(String.format(BF_ACTIONS_ERROR_MSG, caseData.getEthosCaseReference()), errors.get(0));
```

Comment:
May be test the scenario when there are 0 errors

Reply:
Scenario created to test 0 errors